testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveHiddenFiles.java

```diff
@@ -0,0 +1,105 @@
+
+public class TestHiveHiddenFiles
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @Test
+    public void testSelectFromTableContainingHiddenFiles()
+            throws Exception
+    {
+        String tableName = "test_table_hidden_files" + randomNameSuffix();
+        onTrino().executeQuery("CREATE TABLE " + tableName + " (col integer)");
+
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 1");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 2");
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 3");
+
+        List<QueryAssert.Row> tableRows = List.of(row(1), row(2), row(3));
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+
+        String tableLocation = getTablePath(tableName);
+        boolean useDotPrefix = true;
+        // Rename the table files to Hive hidden files (prefixed by `.` or `_` characters)
+        for (String filename : hdfsClient.listDirectory(tableLocation)) {
+            renameFile(tableLocation, filename, (useDotPrefix ? '.' : '_') + filename);
+            useDotPrefix = !useDotPrefix;
+        }
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).hasNoRows();
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).hasNoRows();
+
+        onTrino().executeQuery(format("DROP TABLE IF EXISTS " + tableName));
+    }
+
+    @Test
+    public void testSelectFromTableContainingFilenamesWithUnderscore()
+            throws Exception
+    {
+        String tableName = "test_table_visible_underscore_files" + randomNameSuffix();
+        onTrino().executeQuery("CREATE TABLE " + tableName + " (col integer)");
+
+        onTrino().executeQuery("INSERT INTO " + tableName + " VALUES 1");
+
+        List<QueryAssert.Row> tableRows = List.of(row(1));
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+
+        String tableLocation = getTablePath(tableName);
+        // Prefix the table files with `f_` which should still keep them visible to Hive
+        for (String filename : hdfsClient.listDirectory(tableLocation)) {
+            // As long as the file is not hidden (starting with `.` or `_`), it should not be ignored by Hive
+            renameFile(tableLocation, filename, "f_" + filename);
+        }
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(tableRows);
+
+        onTrino().executeQuery(format("DROP TABLE IF EXISTS " + tableName));
+    }
+
+    private void renameFile(String tableLocation, String filename, String newFilename)
```

Comment:
nit: This method logic doesn't depend on "table". I would rename tableLocation to directoryPath or something.